include/pybind11/pytypes.h

```diff
@@ -303,7 +303,9 @@ class handle : public detail::object_api<handle> {
             stderr,
             "%s is being called while the GIL is either not held or invalid. Please see "
             "https://pybind11.readthedocs.io/en/stable/advanced/"
-            "misc.html#common-sources-of-global-interpreter-lock-errors for debugging advice.\n",
+            "misc.html#common-sources-of-global-interpreter-lock-errors for debugging advice.\n"
+            "If you are absolutely sure there is no bug, you can #define PYBIND11_NO_GIL_CHECKS "
```

Comment:
If you are convinced this is the only alternative, you can `#define PYBIND11_NO_ASSERT_GIL_HELD_INCREF_DECREF` to disable this check. In that case you have to ensure this `#define` is consistently used for all translation units linked into a given pybind11 extension, otherwise there will be ODR violations.

Reply:
Fixed! Good point about the ODR issues.